### PR TITLE
Write telemac format

### LINKED ISSUE #202: Add support to write to SLF (Telemac) format

### PR DESCRIPTION
**Pull request by vcloarec**
fix #202

### CODE REVIEW

File: mdal/frmts/mdal_selafin.cpp [hunk 1]
```diff
@@ -42,33 +49,359 @@ void MDAL::SerafinStreamReader::initialize( const std::string &fileName )
...
+      else
+      {
+        dataset->setXVariableIndex( nName );
+      }
+
+    }
+  }
+
+  // now calculate statistics
+  for ( auto group : groupsInOrder )
```
Comment (PeterPetrik) [inline mdal/frmts/mdal_selafin.cpp:387]: please do not use `auto`

File: mdal/frmts/mdal_selafin.cpp [hunk 2]
```diff
@@ -42,33 +49,359 @@ void MDAL::SerafinStreamReader::initialize( const std::string &fileName )
...
+  {
+    for ( auto dataset : group->datasets )
```
Comment (PeterPetrik) [inline mdal/frmts/mdal_selafin.cpp:389]: please do not use `auto`

File: mdal/frmts/mdal_selafin.cpp [hunk 3]
```diff
@@ -42,33 +49,359 @@ void MDAL::SerafinStreamReader::initialize( const std::string &fileName )
...
+      MDAL::Statistics stats = MDAL::calculateStatistics( dataset );
+      dataset->setStatistics( stats );
+    }
+
+    MDAL::Statistics stats = MDAL::calculateStatistics( group );
+    group->setStatistics( stats );
+  }
+
+  // As everything seems to be ok (no exception thrown), push the groups in the mesh
+  for ( auto group : groupsInOrder )
```
Comment (PeterPetrik) [inline mdal/frmts/mdal_selafin.cpp:400]: please do not use `auto`

File: mdal/frmts/mdal_selafin.cpp [hunk 4]
```diff
@@ -258,478 +582,167 @@ std::streampos MDAL::SerafinStreamReader::passThroughDoubleArray( size_t size )
...
+  try
+  {
+    saveDatasetGroupOnFile( group );
+    return false;
+  }
+  catch ( MDAL_Status error )
+  {
+    MDAL::Log::error( error, name(), "error occurred" );
+    return true;
+  }
```
Comment (PeterPetrik) [inline mdal/frmts/mdal_selafin.cpp:706]: can this happen at all?

File: mdal/frmts/mdal_selafin.hpp [hunk 5]
```diff
@@ -21,20 +21,50 @@ namespace MDAL
 {
   /**
    * This class is used to read the selafin file format.
-   * The file is opened with initialize() and stay opened until this object is destructed
+   * The file is opened with initialize() and stay opened until this object is destroyed
    *
    * \note SerafinStreamReader object is shared between different datasets, with the mesh and its iterators.
    *       As SerafinStreamReader is not thread safe, it has to be shared in the same thread.
    *
    * Each record of the stream has a int (4 bytes) at the beginning and the end of the record, this int is the size of the record
```
Comment (PeterPetrik) [inline mdal/frmts/mdal_selafin.hpp:29]: maybe you can improve a bit docs here ? and also this one for users https://github.com/lutraconsulting/MDAL/blob/master/docs/source/drivers/selafin.rst

File: mdal/frmts/mdal_selafin.cpp (hunk 4 above)
Comment (vcloarec) [reply mdal/frmts/mdal_selafin.cpp:706]: good point...

**Pull request merged**